### ISSUE TITLE
[Java] Implement new Concept Exercise: classes

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1785]
This issue describes how to implement the `classes` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of exercise is to learn how to define a class and its fundamental parts.

## Learning objectives

- Know what classes are.
- Know what encapsulation is.
- Know what fields are.
- Know how to create an object.
- Know how to update state through methods.
- Know about the `void` type.

## Out of scope

- Reference equality.
- Reference parameter passing.
- Constructors.
- Interfaces.
- Inheritance.
- Properties.
- Destructors.
- Method overloading.
- Pattern matching.

## Concepts

- `classes`: know what classes are; know what encapsulation is; know what fields are; know how to create an object; know how to update state through methods; know about the `void` type.

## Prequisites

- `basics`: know how to define a basic class with basic methods.
- `strings`: know how to do basic string interpolation.
- `numbers`: know how to compare numbers.
- `conditionals`: know how to do conditional logic.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer)

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer)

## Implementing
To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by mikedamay** [comment]
Closed with PR #2179
